prompting/datasets/wiki.py

```diff
@@ -1,90 +1,64 @@
-# The MIT License (MIT)
-# Copyright © 2024 Yuma Rao
-# Copyright © 2023 Opentensor Foundation
-
-# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
-# documentation files (the “Software”), to deal in the Software without restriction, including without limitation
-# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
-# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
-
-# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
-# the Software.
-
-# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
-# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
-# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-# DEALINGS IN THE SOFTWARE.
-
 import re
 import sys
 import random
-import datetime
 import bittensor as bt
-import wikipedia as wiki
-from typing import Dict, Union, List, Tuple
+import wikipedia
 from queue import Queue, Full, Empty
 from functools import lru_cache
-from .base import Dataset
-from ..selector import Selector
-
+from prompting.datasets.base import BaseDataset
+from prompting.datasets.base import Context
+from typing import ClassVar
+from typing import Optional
+from pydantic import model_validator, ConfigDict
 
 # Create a queue called CACHED_ARTICLES to store wikipedia articles that have been fetched
-CACHED_ARTICLES = Queue(maxsize=300)
+CACHED_ARTICLES: Queue[Context] = Queue(maxsize=300)
 
 
 # speed up page loading
 @lru_cache(maxsize=1000)
-def _get_page(
-    title, pageid=None, auto_suggest=False, redirect=True, seed=None
-) -> wiki.WikipediaPage:
+def _get_page(title, pageid=None, auto_suggest=False, redirect=True, seed=None) -> wikipedia.WikipediaPage:
```

Comment:
we should define types for these parameters.  e.g.

```py
from typing import Optional

def _get_page(title: str, pageid: Optional[str] = None, auto_suggest: Optional[bool] = False, ...) -> wikipedia.WikipediaPage:
```

NOTE: This is general comment that applies across all files.